grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -2682,6 +2685,11 @@ class InventoryService implements ApplicationContextAware {
 
                         }
 
+                        row.isNewExpirationDate = false
```

Comment:
Why not just: 
`row.isNewExpirationDate = inventoryItem && expirationDate != inventoryItem.expirationDate`